ember-resources/src/util/function.ts

```diff
@@ -92,43 +93,53 @@ export function trackedFunction<Return>(...passedArgs: UseFunctionArgs<Return>)
     fn = passedArgs[2];
   }
 
-  return resource<State<Return>>(context, (hooks) => {
-    let state = new State(initialValue);
+  return resource<TrackedFunctionProperty<Return>>(context, (hooks) => {
+    return new TrackedFunctionProperty(fn, hooks, initialValue);
```

Comment:
Changing the return type is a bit of a breaking change -- any chance you can combine the two classes, in in the `execute` method, "reset" the state values?